actions/generate-schedule/scheduler/scheduler_test.go

```diff
@@ -0,0 +1,199 @@
+package scheduler
+
+import (
+	"github.com/ghodss/yaml"
+	"github.com/golang/protobuf/proto"
+	"github.com/spinnaker/rotation-scheduler/actions/generate-schedule/users"
+	"github.com/spinnaker/rotation-scheduler/proto/schedule"
+	"testing"
+	"time"
+)
+
+func TestNewScheduler(t *testing.T) {
+	if _, err := NewScheduler(nil, 1); err == nil {
+		t.Error("want error on nil userSource and didn't get one.")
+	}
+
+	if _, err := NewScheduler(users.NewStaticSource("foo"), 0); err == nil {
+		t.Errorf("want error on invalid shift duration, and didn't get one.")
+	}
+}
+
+func TestSchedule(t *testing.T) {
+	s, err := NewScheduler(users.NewStaticSource("first", "second", "third"), 1)
+	if err != nil {
+		t.Fatalf("error creating scheduler: %v", err)
+	}
+
+	sched1 := &schedule.Schedule{
+		Shifts: []*schedule.Shift{
+			{
+				StartDate: "Wed 01 Jan 2020",
+				User:      "first",
+			},
+			{
+				StartDate: "Thu 02 Jan 2020",
+				User:      "second",
+			},
+			{
+				StartDate: "Fri 03 Jan 2020",
+				User:      "third",
+			},
+		},
+	}
+
+	for _, tc := range []struct {
+		desc    string
+		start   time.Time
+		stop    time.Time
+		wantErr bool
+		want    *schedule.Schedule
+	}{
+		{
+			desc:    "happy path",
+			start:   time.Date(2020, 01, 01, 0, 0, 0, 0, time.UTC),
+			stop:    time.Date(2020, 01, 04, 0, 0, 0, 0, time.UTC),
+			wantErr: false,
+			want:    sched1,
```

Comment:
Kind of a nit, but just wondering why this `want` is pulled into its own variable and the next want is defined inline?

Reply:
The original thought was for brevity, but it I think it looks good inline. Fixed.